devito/function_manager.py

```diff
@@ -74,6 +90,45 @@ def generate_function_body(self, function_descriptor):
         statements.append(cgen.Statement("return 0"))
         return cgen.Block(statements)
 
+    def _append_main_at_function(self, function_descriptor):
+        """Appends main at function to the list function desc. Works with assumption that we are tuning first function.
+        Args:
+            function_descriptor (FunctionDescriptor): function descriptor which has at main function flag set
+        """
+
+        statements = []  # statements for cgen.block
+        pnames = []  # parameter names
+        main_fd = FunctionDescriptor("main")
+        main_fd.return_type = "int"
+
+        # allocates the space for matrix'es
+        # Note currently auto tunes only the first function in function descriptors. If scope is larger. Extend
+        for param in function_descriptor.matrix_params:
+            array_size_str = ""
+            for shape in param["shape"]:
+                array_size_str += "%s * " % shape
+
+            ptype = cgen.dtype_to_ctype(param['dtype'])
+            pname = param["name"] + "_vec"
+            pnames.append(pname)
+
+            # Produces similar str: double* m_vec =(double*)malloc(336*336*336*sizeof(double))
+            allocation_str = "%s* %s = (%s*)malloc(%ssizeof(%s))" % (ptype, pname, ptype, array_size_str, ptype)
+            statements.append(cgen.Statement(allocation_str))
```

Comment:
This is still not doing any initialisation. Did we get a final word on whether this is what we want?